### PR TITLE
ci: Add GitHub Workflow (#1879)

### PR DESCRIPTION
**Pull request by jcubic**
* chore(deps-dev): bump express from 4.17.1 to 4.19.2 (#1891)

Bumps [express](https://github.com/expressjs/express) from 4.17.1 to 4.19.2.
- [Release notes](https://github.com/expressjs/express/releases)
- [Changelog](https://github.com/expressjs/express/blob/master/History.md)
- [Commits](https://github.com/expressjs/express/compare/4.17.1...4.19.2)

---
updated-dependencies:
- dependency-name: express dependency-type: indirect ...

* fix: handle no-refs response from git-upload-pack for git > 2.41.0 (#1862)

* ci: Add GitHub Workflow

* Prevent ERR_OSSL_EVP_UNSUPPORTED on Node >= 17

---------

## I'm fixing a bug or typo

- [ ] if this is your first time contributing, run `npm run add-contributor` and follow the prompts to add yourself to the README
- [ ] squash merge the PR with commit message "fix: [Description of fix]"

## I'm adding a parameter to an existing command X:

- [ ] add parameter to the function in `src/api/X.js` (and `src/commands/X.js` if necessary)
- [ ] document the parameter in the JSDoc comment above the function
- [ ] add a test case in `__tests__/test-X.js` if possible
- [ ] if this is your first time contributing, run `npm run add-contributor` and follow the prompts to add yourself to the README
- [ ] squash merge the PR with commit message "feat(X): Added 'bar' parameter"

## I'm adding a new command:

- [ ] add as a new file in `src/api` (and `src/commands` if necessary)
- [ ] add command to `src/index.js`
- [ ] update `__tests__/test-exports.js`
- [ ] create a test in `src/__tests__`
- [ ] document the command with a JSDoc comment
- [ ] add page to the Docs Sidebar `website/sidebars.json`
- [ ] add page to the v1 Docs Sidebar `website/versioned_sidebars/version-1.x-sidebars.json`
- [ ] if this is your first time contributing, run `npm run add-contributor` and follow the prompts to add yourself to the README
- [ ] squash merge the PR with commit message "feat: Added 'X' command"

### CODE REVIEW

File: src/wire/parseRefsAdResponse.js [hunk 1]
```diff
@@ -41,14 +41,17 @@ export async function parseRefsAdResponse(stream, { service }) {
...
+  if (firstRef !== '0000000000000000000000000000000000000000 capabilities^{}') {
+    const [ref, name] = splitAndAssert(firstRef, ' ', ' ')
+    refs.set(name, ref)
+    while (true) {
+      const line = await read()
+      if (line === true) break
+      if (line !== null) {
+        const [ref, name] = splitAndAssert(line.toString('utf8'), ' ', ' ')
+        refs.set(name, ref)
+      }
```
Comment (seanpoulter) [inline src/wire/parseRefsAdResponse.js:54]: I believe this is from another commit. Is it worth a rebase or merge to keep it separate?

**Comment by seanpoulter** [comment]
That seems to work well, eh? We could fix the Node 16 warning from Codespell in another PR.

File: src/wire/parseRefsAdResponse.js (hunk 1 above)
Comment (jcubic) [reply src/wire/parseRefsAdResponse.js:54]: Yea, it was the commit to handle empty repositories.

Comment (jcubic) [reply src/wire/parseRefsAdResponse.js:54]: I will need to redo the merge locally, because GitHub required to squash the whole history.

Comment (jcubic) [reply src/wire/parseRefsAdResponse.js]: Created second PR https://github.com/isomorphic-git/isomorphic-git/pull/1902

**Pull request closed without merging**